script.js

```diff
@@ -1 +1,406 @@
+    // Store the current suit in a variable
+    let currentSuit = value.suitsShape;
+    let currentSymbol = value.suitsSymbol;
+    let currentColour = value.suitsColour;
+
+    // Loop from 1 to 13 to create all cards for a given suit
+    // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12.
+    // This is an example of a loop without an array.
+    for (let rankCounter = 1; rankCounter <= 13; rankCounter++) {
+      // By default, the card name is the same as rankCounter
+      let cardName = rankCounter;
+      let displayName = rankCounter;
+
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name, set cardValue to 10 (for jack, queen, king) or 11 (for ace)
+      if (rankCounter == 1) {
+        cardName = "ace";
+        displayName = "A";
+      } else if (rankCounter == 11) {
+        cardName = "jack";
+        displayName = "J";
+      } else if (rankCounter == 12) {
+        cardName = "queen";
+        displayName = "Q";
+      } else if (rankCounter == 13) {
+        cardName = "king";
+        displayName = "K";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      const card = {
+        suitSymbol: currentSymbol,
+        suit: currentSuit,
+        name: cardName,
+        display: displayName,
+        colour: currentColour,
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      newDeck.push(card);
+      newDeck.push(card);
+    }
+  }
+  // Return the completed card deck
+  return newDeck;
+};
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+const getRandomIndex = (max) => Math.floor(Math.random() * max);
+
+// Shuffle an array of cards
+const shuffleCards = (cards) => {
+  // Loop over the card deck array once
+  for (let currentIndex = 0; currentIndex < cards.length; currentIndex += 1) {
+    // Select a random index in the deck
+    const randomIndex = getRandomIndex(cards.length);
+    // Select the card that corresponds to randomIndex
+    const randomCard = cards[randomIndex];
+    // Select the card that corresponds to currentIndex
+    const currentCard = cards[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cards[currentIndex] = randomCard;
+    cards[randomIndex] = currentCard;
+  }
+  // Return the shuffled deck
+  return cards;
+};
+
+//for cardBack design and flip if not matched
+const cardBackDeco = () => {
+  squareDeco = document.createElement("img");
+  squareDeco.classList.add("card-deco");
+  squareDeco.src = "assets/images/xmas.svg";
+};
+
+//for game info text
+const output = (message) => {
+  gameInfo.innerText = message;
+};
+
+// for timer text
+const timeOutput = () => {
+  timeRemaining.innerHTML = `Time \n
+  ${minute}:${sec.toString().padStart(2, "0")}`;
+};
+
+const scoreOutput = () => {
+  scoreElement.innerText = `Score ${score}`;
+};
+
+//timer & reset function
+const countDown = () => {
+  handle = setInterval(function () {
```

Comment:
I guess this is a very minor issue, but consistency is a highly valuable trait, if we have been using arrow functions throughout, it might be a good idea to use an arrow function here as well